text/0017-gas-futures.md

```diff
@@ -0,0 +1,147 @@
+# Gas futures
+
+## Summary
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0017](https://github.com/casperlabs/ceps/pull/0017)
+
+A certain percentage of future block space is reserved and auctioned off. Said block space can only be filled by transactions signed by gas futures holders.
+
+## Motivation
+
+[motivation]: #motivation
+
+Gas prices are known to be subject to high volatility both in the short term and the long term. This is due to a combination of factors such as inelastic supply, daily demand cycle and shifting trends. CasperLabs sees these issues with volatility as a key barrier to the adoption of public blockchain infrastructure by businesses. It is also recognized that price stability alone is not enough, and scalability is a core concern.
+
+During times of high demand for a blockchain, one can observe prices change up to 10-100x in a matter of hours. A floating gas price and the absence of a reference point exacerbates the situation. Such volatility makes it difficult for dapp providers and users to budget for their expenses.
+
+We propose gas futures to enable users to hedge themselves against future gas price uncertainty.
+
+## Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+Prior to this proposal, each block had a `BLOCK_GAS_LIMIT` which determined the throughput of the blockchain. We introduce a protocol parameter `FUTURES_GAS_LIMIT <= BLOCK_GAS_LIMIT`. This portion of the block space will be exclusive to gas futures holders, that is
+
+```python
+X <= BLOCK_GAS_LIMIT - FUTURES_GAS_LIMIT
+```
+
+where `X` is the total gas usage of transactions signed by users that do not hold gas futures. Blocks that do not fulfill this condition are deemed invalid.
+
+Gas futures are designated for consecutive periods of fixed length. In this proposal, we will focus on daily gas futures, but the same instrument can be extended to designate weekly, monthly, and so on. Gas futures are represented by tokens which are auctioned off continuously, some fixed duration preceding the futures date. (In this proposal, "futures" and "futures tokens" are used interchangeably.) These tokens will be designated in the format `GASYYYYMMDD`, and will be sold off in an on-chain auction `AUCTION_PERIOD` days before the futures date. In this proposal, we will assume `AUCTION_PERIOD = 180`, i.e. 6 months.
```

Comment:
Let's make sure we call it `GAS_AUCTION_PERIOD` in the code to disambiguate. There's also the validator slot auction (which even has its own `AUCTION_DELAY`).
For the CEP it's fine, of course.

Reply:
Done